src/domain/community/inviteContributors/InviteVCsDialog.tsx

```diff
@@ -109,6 +117,11 @@ const InviteVCsDialog = ({ open, onClose }: InviteContributorDialogProps) => {
     return onAccount?.find(c => c.id === id) || inLibrary?.find(c => c.id === id);
   };
 
+  // remove the duplicates from available onAccount in the inLibrary list
+  const filteredInLibrary = useCallback(() => {
```

Comment:
While this is perfectly working implementation of this method, I think we can lower down from `O(n * m)` to `O(n + m)` (`n` is `inLibrary`'s length, `m` is `onAccount`'s length) by doing so:
```
const filteredInLibrary = useCallback(() => {
  if (!inLibrary) return [];

  // Creating a Map for quick access to elements in onAccount by id. This allows the check if id is present to be lowered down to O(1), e.g., reduces execution time from quadratic to linear due to the use of a hash table (Map).
  const accountIds = new Map(onAccount?.map(member => [member.id, true]));

  // inLibrary filtering where we check if id is present in accountIds
  return inLibrary.filter(vc => !accountIds.has(vc.id));
}, [inLibrary, onAccount]);
```

Reply:
We can do it. I used to think more about that in the past, but this syntaxis sugar is mainly used in this project.
The VCs on the account are not more then 3 but the library ones could grow so it makes sense, thanks.

Reply:
I've implemented it with Set and extracted it as a util for getting the unique items, we're using such a comparison on many places.